### PR TITLE
refactor: use common-library securityContext helper

### PR DESCRIPTION
**Pull request by marcsanmi**
Relates to https://github.com/newrelic/nri-kube-events/pull/106.

This PR is a breaking change as we chop the usage of `runAsUser`.

Defaults are being kept the same.

### CODE REVIEW

File: charts/nri-kube-events/templates/_helpers_compatibility.tpl [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{{/*
+Returns a dictionary with legacy runAsUser config
+*/}}
+{{- define "newrelic.compatibility.securityContext" -}}
+{{- if  .Values.runAsUser -}}
```
Comment (gsanchezgavier) [inline charts/nri-kube-events/templates/_helpers_compatibility.tpl:5]: could you add a unit test to check for the compatibility layer ?

File: charts/nri-kube-events/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{{/* vim: set filetype=mustache: */}}
+
+{{- define "nri-kube-events.securityContext.pod" -}}
+{{- if include "newrelic.common.securityContext.pod" . -}}
+{{- include "newrelic.common.securityContext.pod" . -}}
+{{- else if include "newrelic.compatibility.securityContext" . -}}
+{{- include "newrelic.compatibility.securityContext" . -}}
```
Comment (gsanchezgavier) [inline charts/nri-kube-events/templates/_helpers.tpl:7]: shouldn't this definition return an string ? something like this 

```
runAsUser: {{ include "newrelic.compatibility.securityContext" . }}
```

File: charts/nri-kube-events/templates/_helpers_compatibility.tpl (hunk 1 above)
Comment (marcsanmi) [reply charts/nri-kube-events/templates/_helpers_compatibility.tpl:5]: Added them here f39d852f58ab2cb7a03122d1e02722feec1ca659

File: charts/nri-kube-events/templates/_helpers.tpl (hunk 2 above)
Comment (marcsanmi) [reply charts/nri-kube-events/templates/_helpers.tpl]: I think that as we are using `{{ dict "runAsUser" .Values.runAsUser | toYaml }}`  to create the value in the compatibility layer, we are good. However, I removed the right dash.

File: charts/nri-kube-events/tests/deployment_test.yaml [hunk 3]
```diff
@@ -44,6 +44,78 @@ tests:
...
+    set:
+      cluster: test-cluster
+      runAsUser: 1001
+    asserts:
+      - equal:
+          path: spec.template.spec.securityContext.runAsUser
+          value: 1001
+      - equal:
+          path: spec.template.spec.securityContext.runAsNonRoot
+          value: true
```
Comment (kang-makes) [inline charts/nri-kube-events/tests/deployment_test.yaml:118]: There are missing tests that test that the compatibility layer overrides the values that came from the common library.

**Review by kang-makes — COMMENTED** [review]
There is a couple of things that can be improved or that can open a discussion about the order of merging values.

It has good shape tho.

File: charts/nri-kube-events/tests/deployment_test.yaml (hunk 3 above)
Comment (marcsanmi) [reply charts/nri-kube-events/tests/deployment_test.yaml]: Updated in 49fa2046a327f65fc6756382dd96ae5a52cb51a0.

File: charts/nri-kube-events/templates/deployment.yaml [hunk 4]
```diff
@@ -139,3 +140,8 @@ spec:
       dnsConfig:
         {{- . | nindent 8 }}
       {{- end }}
+      hostNetwork: {{ include "newrelic.common.hostNetwork.value" . }}
+      {{- with include "newrelic.common.dnsConfig" . }}
+      dnsConfig:
+        {{- . | nindent 8 }}
+      {{- end }}
```
[the remark this answers is not in the document]
Comment (marcsanmi) [reply charts/nri-kube-events/templates/deployment.yaml]: Something messed up when rebasing, fixed. Thanks!

File: charts/nri-kube-events/templates/_helpers.tpl [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+{{- if include "newrelic.common.securityContext.pod" . -}}
+{{- include "newrelic.common.securityContext.pod" . -}}
+{{- else if include "newrelic.compatibility.securityContext" . -}}
+{{- include "newrelic.compatibility.securityContext" . }}
+runAsNonRoot: true
+{{- else -}}
+runAsUser: 1000
+runAsNonRoot: true
+{{- end -}}
+{{- end -}}
```
[the remark this answers is not in the document]
Comment (marcsanmi) [reply charts/nri-kube-events/templates/_helpers.tpl]: Updated in 148f787806286588deff69dcfbb5e335920621dd. Thanks for the example 👌

File: charts/nri-kube-events/tests/deployment_test.yaml [hunk 6]
```diff
@@ -44,6 +44,78 @@ tests:
           content:
             name: NRIA_PROXY
             value: "https://proxy:1234"
+  - it: set to defaults when no podSecurityContext set
```
[the remark this answers is not in the document]
Comment (marcsanmi) [reply charts/nri-kube-events/tests/deployment_test.yaml]: Updated.

File: charts/nri-kube-events/templates/_helpers_compatibility.tpl [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+{{ dict "runAsUser" .Values.runAsUser | toYaml }}
```
[the remark this answers is not in the document]
Comment (marcsanmi) [reply charts/nri-kube-events/templates/_helpers_compatibility.tpl]: Done in 148f787806286588deff69dcfbb5e335920621dd.

File: charts/nri-kube-events/templates/_helpers_compatibility.tpl [hunk 8]
```diff
@@ -0,0 +1,8 @@
+{{/*
+Returns a dictionary with legacy runAsUser config
+*/}}
```
[the remark this answers is not in the document]
Comment (marcsanmi) [reply charts/nri-kube-events/templates/_helpers_compatibility.tpl]: Done in 148f787806286588deff69dcfbb5e335920621dd.

**Review by kang-makes — DISMISSED** [review]
Left a few nits but LGTM!

**Review by kang-makes — APPROVED** [review]
👍

**Pull request merged**